multi_physics/QED/kokkos_example/kokkos_example_commons.hpp

```diff
@@ -0,0 +1,351 @@
+    * are forwarded to Kokkos::vector constructor.
+    *
+    * @tparam Args the constructor arguments
+    */
+    template<typename... Args>
+    KokkosVectorWrapper(Args&&... args) : KV(std::forward<Args>(args)...)
+    {}
+
+    /**
+    * This function may be called in some places inside picsar_table classes.
+    * It forces a deep copy of the CPU data to the GPU
+    *
+    */
+    void pxr_sync()
+    {
+        Kokkos::deep_copy(KV::d_view, KV::h_view);
+    }
+
+    /**
+    * This function returns a point to the raw data inside Kokkos::vector.
+    * It is called to build table_views.
+    *
+    * @return a pointer to the raw vector data
+    */
+    const Real* data() const
+    {
+        return KV::d_view.data();
+    }
+
+};
+
+/**
+* An auxiliary function to call typeid(T).name()
+*
+* @tparam T a typename
+* @return the name of T as a string
+*/
+template<typename T>
+std::string get_type_name()
+{
+    return typeid(T).name();
+}
+
+/**
+* Initializes a 2D Kokkos::View with N, ParticleData<Real>::num_components
+* elements. The values inside the View are initialized randomly between min_val
+* and max_val.
+*
+* @tparam Real the floating point type to be used
+* @param[in] label the Kokkos::View label
+* @param[in] N the number of particles
+* @param[in] min_val the minimum value to initialize the content of the Kokkos::View
+* @param[in] max_val the maximum value to initialize the content of the Kokkos::View
+* @param[in] N the number of particles
+* @param[in,out] rand_pool a random pool
+* @return the multi-component view
+*/
+template <typename Real>
+auto init_multi_comp_view_with_random_content(
+    const std::string& label,
+    const Real min_val, const Real max_val,
+    const int N,
+    Kokkos::Random_XorShift64_Pool<>& rand_pool)
+{
+    auto view = Kokkos::View<Real * [ParticleData<Real>::num_components]>{label, static_cast<unsigned int>(N)};
+
+    Kokkos::fill_random(view, rand_pool, min_val, max_val);
+
+    return view;
+}
+
+/**
+* Initializes a 1D Kokkos::View with N elements.
+* The values inside the View are initialized randomly between min_val
+* and max_val.
+*
+* @tparam Real the floating point type to be used
+* @param[in] label the Kokkos::View label
+* @param[in] N the number of particles
+* @param[in] min_val the minimum value to initialize the content of the Kokkos::View
+* @param[in] max_val the maximum value to initialize the content of the Kokkos::View
+* @param[in] N the number of particles
+* @param[in,out] rand_pool a random pool
+* @return the multi-component view
+*/
+template <typename Real>
+auto init_view_with_random_content(
+    const std::string& label,
+    const Real min_val, const Real max_val,
+    const unsigned int N,
+    Kokkos::Random_XorShift64_Pool<>& rand_pool)
+{
+    auto view = Kokkos::View<Real *>{label, N};
+    Kokkos::fill_random(view, rand_pool, min_val, max_val);
+    return view;
+}
+
+/**
+* Initializes particle data
```

Comment:
We could potentially describe this function in more details.

Reply:
I agree... I've added a more in-depth description